File: eojeen/Stack.java
import java.util.EmptyStackException;

public class Stack {

	private Node top;
	 
    public Stack() {
        this.top = null;
    }
 
    public class Node {
        private String data;
        private Node nextNode;
 
        public Node(String data) {
            this.data = data;
            this.nextNode = null;
        }
 
    }
 
    public String push(String data) {
        Node node = new Node(data);
        node.nextNode = top;
        top = node;
        
        return data;
    }
 
    public String pop() {
        String data = peek();
        top = top.nextNode;        
        return data;
    }
 
    public String peek() {
        if (top == null)
            throw new EmptyStackException();
        return top.data;
    }
 
    public boolean empty() {
        return top == null;
    }
 
    public int search(Object o) {
        int index = 1;
        Node tempTop = top;
        
        while (tempTop != null) {
            if (tempTop.data.equals(o))
                return index;                
            tempTop = tempTop.nextNode;
            
            index++;
        }
 
        return -1;
    }

}

File: Nayoung/week4/stack.cpp
#include <iostream>
using namespace std;

struct Node{
	int data;
	Node* next;
};

Node* Top = NULL;

bool IsEmpty() {
	if (Top == NULL) return true;
	else return false;
}

void Push(int data) {
	Node* p = new Node;
	p->data = data;
	p->next = Top;
	Top = p;
}

int Pop() {
	if (!IsEmpty()) {
		Node* temp = Top;
		int data = temp->data;
		Top = temp->next;
		delete(temp);
		return data;
	}
}

void ShowStack() {
	if (!IsEmpty()) {
		Node* temp = Top;
		while (temp != NULL) {
			cout << temp->data << ' ';
			temp = temp->next;
		}
		cout << '\n';
	}
}

int main() {
	Push(10);
	Push(20);
	Push(30);
	Push(40);

	ShowStack();
	Pop();

	Push(50);
	Push(60);
	Push(70);

	ShowStack();
	Pop();
	Pop();
	ShowStack();

	return 0;
}
